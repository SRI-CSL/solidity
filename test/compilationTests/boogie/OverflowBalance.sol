pragma solidity ^0.4.24;

/**
 * @notice invariant bal == this.balance
 */
contract Overflow {
    uint256 bal;

    constructor() {
        bal = 0;
    }

    function receive() public payable {
        bal += msg.value; // Should not overflow
    }
}