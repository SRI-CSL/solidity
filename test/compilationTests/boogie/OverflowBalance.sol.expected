Verifying Overflow::[constructor]: OK
Verifying Overflow::receive: OK
No errors found.
